gato/gatos/ABaseGato.py

```diff
@@ -427,3 +427,16 @@ def add_energy(self, amount: float):
             self.energy = self.max_energy
         elif self.energy <= 0.0:
             self.energy = 0.0
+
+    def set_eidolon(self, value: int):
+        """Setter for a gato's eidolon value.
+        Value must be an integer between 0 and 6 (inclusive).
+        Use this rather than modifying :py:attr:`eidolon` directly.
+        *Can be overriden.* 
+
+        :param amount: New eidolon value to set.
+        :type amount: int
+        """
+        
+        # clamp value to [0, 6]
+        self.eidolon = max(min(value, 6), 0)
```

Comment:
Sticking with `set_eidolon` instead of a `pulled()` for this PR. Reason is as mentioned over on discord 
> But I'm not sure when that (`pulled)` would be called. i.e. when eidolon is increased only? eidolon increase and initial pull? also when 7+ copy is pulled?